docs/source-2.0/aws/rules-engine/library-functions.rst

```diff
@@ -0,0 +1,237 @@
+.. _rules-engine-aws-library-functions:
+
+==================================
+AWS rules engine library functions
+==================================
+
+AWS-specific rules engine library :ref:`functions <rules-engine-endpoint-rule-set-function>`
+make it possible to integrate AWS concepts like `Amazon Resource Names (ARNs)`_
+and `Partitions`_. An additional dependency is required to access these
+functions:
+
+The following example adds ``smithy-aws-rules-engine`` as a Gradle dependency
+to a ``build.gradle.kts`` file:
+
+.. code-block:: kotlin
+
+    dependencies {
+        ...
+        implementation("software.amazon.smithy:smithy-aws-rules-engine:__smithy_version__")
+        ...
+    }
+
+.. _rules-engine-aws-library-awsPartition:
+
+``aws.partition`` function
+==========================
+
+Summary
+    Provides a `Partition structure`_ of content for the
+Argument type
+    * region: ``string``
+Return type
+    ``option<Partition>``
+
+    *Contains the region's partition, or an empty optional if the region is
+    unknown.*
+
+The following example uses ``aws.partition`` to partition information from the
+region in the value of the ``foo`` parameter:
+
+.. code-block:: json
+
+    {
+        "fn": "aws.partition",
+        "argv": [
+            {"ref": "foo"}
+        ]
+    }
+
+
+.. _rules-engine-aws-library-awsPartition-Partition:
+
+-----------------------
+``Partition`` structure
+-----------------------
+
+The ``Partition`` structure is returned from the `aws.partition function`_ when
```

Comment:
Maybe also link this to https://docs.aws.amazon.com/whitepapers/latest/aws-fault-isolation-boundaries/partitions.html